### DISCUSSION TITLE
Apitally - Simple API monitoring & analytics for Starlette apps

### DISCUSSION THREAD

**Discussion by itssimon** [encode/starlette#2630] [Ideas]
Hey Starlette community,

I’d like to introduce you to my indie product [Apitally](https://apitally.io/starlette), a simple API monitoring & analytics tool for Starlette (and also FastAPI).

Apitally provides insights into API traffic, errors and performance, for the whole API, each endpoint and individual API consumers. It also monitors API uptime & availability, alerting users when their API is down.

The big monitoring platforms (Datadog etc.) can be a bit overwhelming & expensive, particularly for simpler use cases. So Apitally’s key differentiators are simplicity & affordability, with the goal to make it as easy as possible for users to understand their APIs usage and performance.

Apitally integrates with Starlette through middleware, which captures request & response metadata (never anything sensitive!) and asynchronously ships it to Apitally’s servers in regular intervals.

Here's a screenshot of the Apitally dashboard:

![Apitally dashboard](https://github.com/user-attachments/assets/188abc71-0492-484a-b49b-2312073d5d31)

If you'd like to try it out, there's a (short) [setup guide for Starlette](https://docs.apitally.io/frameworks/starlette).

Please let me know what you think!

**Comment by sahasrara62** [comment]
looks great, few suggestion from my side
- need colour improvement for colourblinds people
- custom alert and query need to be added
- integration of alerts with 3rd party apps would be great
- graphical data representation can be improved, to get more insight

**Reply by itssimon** [reply]
Thanks so much for taking the time to provide feedback. Highly appreciated and your points are duly noted.
May I ask for an example of how you think the data representation could be improved?

**Reply by itssimon** [reply]
Custom alerts are now available in Apitally 🎉
